virtual-setup/working-with-your-own-data.md

```diff
@@ -0,0 +1,233 @@
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+  - [Things to know before uploading your data:](#things-to-know-before-uploading-your-data)
+  - [Upload data that is online (from a url)](#upload-data-that-is-online-from-a-url)
+  - [Upload data from your own computer](#upload-data-from-your-own-computer)
+  - [Reading the data into your R environment.](#reading-the-data-into-your-r-environment)
+- [Downloading files](#downloading-files)
+- [Installing packages](#installing-packages)
+  - [Finding what packages are installed](#finding-what-packages-are-installed)
+  - [Installing a new package](#installing-a-new-package)
+    - [install.packages()](#installpackages)
+    - [Bioconductor packages](#bioconductor-packages)
+  - [More resources on package installation strategies](#more-resources-on-package-installation-strategies)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+### Upload data that is online (from a url)
+
+If you are retrieving your data from online, perhaps from a publicly available repository, we encourage you to use the terminal command `wget`.
+
+**Step 1)** Go to the Terminal tab in your RStudio session.
+
+![Terminal tab](screenshots/rstudio-session-terminal.png)
+
+**Step 2)** Set up your `wget` command in a script (or notebook) using this template.
+
+The most simple `wget` command just needs the URL to pull the file from.
+
+*Template:*
+```
+wget <URL>
+```
+
+*Specific example:* Here's an example of us downloading a file from ArrayExpress
+```
+wget https://www.ebi.ac.uk/arrayexpress/files/E-GEOD-67851/E-GEOD-67851.processed.1.zip
+```
+
+By default, the file will be saved to the current directory and the file name it had from its origin (so with the above example `E-GEOD-67851.processed.1.zip`).
+
+Likely you will want to be more specific about where you are saving the file to and what you are calling it.
+For that, we can use the `-O`, or `output` option with our `wget` command and specify a file path.  
+
+*Template:*
+```
+wget -O <FILE_PATH_TO_SAVE_TO> <URL>
+```
+
+*Specific example:* Here's an example where we will download that same array express file, but instead save it to the `data` folder and call it `some_array_data.zip`.
+(Best to keep the file extension consistent to avoid troubles!)
+
+```
+wget -O data/some_array_data.zip https://www.ebi.ac.uk/arrayexpress/files/E-GEOD-67851/E-GEOD-67851.processed.1.zip
+```
+
+`-O` is one of many `wget` command options.
+To see the complete list of `wget` options, use the command: `wget -h` in bash.
+
+[See more `wget` examples](https://www.tecmint.com/10-wget-command-examples-in-linux/)
+
+### Upload data from your own computer
+
+If the data you want to use is stored locally on your computer, here's how we recommend uploading it to the RStudio Server.
+
+**Step 1)** We recommend you compress your data folder into a single zip file.
```

Comment:
Note limit for browser uploads to the server is probably somewhere around 1GB, so they may need to split things up, or ask for help.